test/extended/apiserver/tls.go

```diff
@@ -1,31 +1,114 @@
+
+		defer func() {
+			g.By("Cleanup - removing TLS profile")
+
+			cleanupCtx, cancelCleanup := context.WithTimeout(context.Background(), tlsWaitForCleanupDuration)
+			defer cancelCleanup()
+
+			kasStatus, err := operatorClient.OperatorV1().KubeAPIServers().Get(cleanupCtx, "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			currentRevision := kasStatus.Status.LatestAvailableRevision
+
+			removeModernProfilePatch := `[{"op":"remove","path":"/spec/tlsSecurityProfile"}]`
+
+			_, err = configClient.ConfigV1().APIServers().Patch(cleanupCtx, "cluster", types.JSONPatchType, []byte(removeModernProfilePatch), metav1.PatchOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			_, err = waitForKASIncrementedRevision(cleanupCtx, operatorClient, "cluster", currentRevision)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}()
+
+		g.By("Applying a JSON Patch to use the modern TLS profile")
+
+		kasStatus, err := operatorClient.OperatorV1().KubeAPIServers().Get(ctx, "cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		currentKASRevision := kasStatus.Status.LatestAvailableRevision
+
+		addModernProfilePatch := `[{"op":"add","path":"/spec/tlsSecurityProfile","value":{"type":"Modern","modern":{}}}]`
+
+		_, err = configClient.ConfigV1().APIServers().Patch(ctx, "cluster", types.JSONPatchType, []byte(addModernProfilePatch), metav1.PatchOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Waiting for etcd to stabilize")
+
+		podClient := oc.AdminKubeClient().CoreV1().Pods("openshift-etcd")
+		err = library.WaitForPodsToStabilizeOnTheSameRevision(t, podClient, "app=etcd", 5, 24*time.Second, 5*time.Second, 30*time.Minute)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Dialing the API with a minimum TLS version of 1.3")
+
+		_, err = waitForKASIncrementedRevision(ctx, operatorClient, "cluster", currentKASRevision)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		config := &tls.Config{MinVersion: tls.VersionTLS13, MaxVersion: tls.VersionTLS13, InsecureSkipVerify: true}
+
+		host := strings.TrimPrefix(oc.AdminConfig().Host, "https://")
+
+		conn, err := tls.Dial("tcp4", host, config)
+		if err != nil {
+			t.Fatalf("Expected success with TLS 1.3, got %v", err)
+		}
+
+		conn.Close()
+	})
+})
+
 var _ = g.Describe("[sig-api-machinery][Feature:APIServer]", func() {
 	defer g.GinkgoRecover()
 
 	oc := exutil.NewCLI("apiserver")
 
 	g.It("TestTLSDefaults", func() {
-		g.Skip("skipping because it was broken in master")
-
 		t := g.GinkgoT()
 		// Verify we fail with TLS versions less than the default, and work with TLS versions >= the default
 		for _, tlsVersionName := range crypto.ValidTLSVersions() {
 			tlsVersion := crypto.TLSVersionOrDie(tlsVersionName)
 			expectSuccess := tlsVersion >= crypto.DefaultTLSVersion()
 			config := &tls.Config{MinVersion: tlsVersion, MaxVersion: tlsVersion, InsecureSkipVerify: true}
 
+			host := strings.TrimPrefix(oc.AdminConfig().Host, "https://")
```

Comment:
nit: would you mind adding a comment why this is necessary so we don't forget a year from now :)

Reply:
Absolutely!